projects/ui/src/Apis/hooks.ts

```diff
@@ -70,7 +77,41 @@ export function useGetCurrentUser() {
 }
 
 export function useListApis() {
-  return useSwrWithAuth<API[]>("/apis");
+  const res = useSwrWithAuth<API[] | APIProduct[]>("/apis");
+  //
+  // This is a preventative fix for the API list page to not break while it
+  // is being updated to use the new server-side APIProduct grouping.
+  // TODO: Update this when the portal server REST API is updated: https://github.com/solo-io/gloo-mesh-enterprise/issues/10067
+  //
+  let processedAPIs = res.data as API[];
+  if (!!res.data?.length && "apiVersions" in res.data[0]) {
+    //
+    // If res.data is an APIProduct[], convert it to an API[].
+    //
+    let apiProducts = res.data as APIProduct[];
+    processedAPIs = apiProducts.reduce((accum, curProd) => {
+      accum.push(
+        ...curProd.apiVersions.reduce((accum, api) => {
+          accum.push({
```

Comment:
nit: I might be overlooking something, but this just seems like a `map()` with extra steps

Reply:
The response data has 2 levels of arrays, like:
```ts
[
... {
  ...
  apiVersions: [ {...}, ... ]
]
```
so doing a map would end up looking like [ [...], [...] ... ].

But now that you mention it, this actually could be doing a [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap). That would flatten it and end up with one array though I don't think I'll update this now since it's doing the same thing.